### ISSUE TITLE
SVG plots in IPython?

### ISSUE THREAD

**Issue by jabirali** [astoff/comint-mime#20]
First of all, thanks for creating this nice package!

I have been trying it out with `python-mode` via `ipython` on Emacs Mac port (v29.1) using the configuration below and it works quite well. However, the resolution of the Matplotlib plots is unfortunately not great; it might be because I'm using a hiDPI / Retina screen (and I know this is handled differently on different Emacs ports), but the text in the Matplotlib figure is quite blurry.

One solution is to use SVG figures instead of PNG figures for the Matplotlib plots. I believe e.g. VSCode offers this as a setting when working with IPython / Jupyter. For comparison, I have added a screenshot comparing how a Matplotlib figure looks in the Inferior Python buffer with `comint-mime` (top right) vs. how it looks when I run `plt.savefig("test.svg")` and open the corresponding SVG file directly in Emacs `image-mode` (bottom). As you can see, the SVG version has much crisper text.

Is there a way to use SVG figures as a format when working with Matplotlib plots in `comint-mime`?

Screenshot:
<img width="1498" alt="Screenshot 2024-04-24 at 11 20 04" src="https://github.com/astoff/comint-mime/assets/271284/0bc44d46-db0a-4dac-b821-4071ee7399ed">

My configuration:
```emacs-lisp
(use-package comint-mime
  :ensure t
  :hook (inferior-python-mode . comint-mime-setup))

(use-package python
  :custom
  (python-indent-guess-indent-offset t)
  (python-indent-guess-indent-offset-verbose nil)
  (python-shell-interpreter "ipython3")
  (python-shell-interpreter-args "--simple-prompt --classic"))
```

**Comment by jabirali** [comment]
For reference: I tested VSCode's "Interactive Python" interface to the IPython / Jupyter interpreter. There, the following IPython magic command produced the same level of improvement by switching to SVG plots:
```
%config InlineBackend.figure_formats = ['svg']
```
However, when I run the same command in IPython running via `comint-mime`, the Matplotlib plots simply disappear. I found the cause: It appears to happen due to [the MIME_TYPE dict](https://github.com/astoff/comint-mime/blob/9d4fe3694d1cb558e26cffb4f92f7d2ffc40498d/comint-mime.py#L22-L27) not listing SVG explicitly.

When I add `"image/svg+xml": str.encode` to that dict, it seems that I get the crisp SVG plots also with `comint-mime` after the above IPython magic:

<img width="1050" alt="Screenshot 2024-04-24 at 16 46 40" src="https://github.com/astoff/comint-mime/assets/271284/68d24788-3d65-42ad-969d-6fb0ff03ab79">

This seems to resolve it, so I can just open a PR :).
